polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -997,8 +988,14 @@ impl State {
 	}
 
 	fn cleanup_assignments_gathering_timestamp(&mut self, keep_greater_than: BlockNumber) {
-		self.per_block_assignments_gathering_times
-			.retain(|block_number, _| *block_number > keep_greater_than)
+		while let Some((block_number, _)) = self.per_block_assignments_gathering_times.peek_oldest()
+		{
+			if *block_number < keep_greater_than {
```

Comment:
the name `keep_greater_than` implies it should be `<=` instead of `<`, but so either change the name or the operand :)